src/mlpack/methods/ann/layer/nearest_interpolation_impl.hpp

```diff
@@ -65,7 +126,7 @@ void NearestInterpolation<InputDataType, OutputDataType>::Forward(
   assert(inRowSize >= 2);
   assert(inColSize >= 2);
 
-  arma::cube inputAsCube(const_cast<arma::Mat<eT>&>(input).memptr(),
+  arma::cube inputAsCube(const_cast<MatType&>(input).memptr(),
       inRowSize, inColSize, depth * batchSize, false, false);
   arma::cube outputAsCube(output.memptr(), outRowSize, outColSize,
       depth * batchSize, false, true);
```

Comment:
We need to replace the `.memptr()` calls and replace them with a MakeAlias, so basically the idea is that we make an alias of the matrix and then we pass it to the Cube.
There is a function `MakeAlias()` in the `core/math/`, also we are using that in several places in the Neural Network code base.